### ISSUE TITLE
Migrate to using scss everywhere

### ISSUE THREAD

**Issue by ermish** [ourjapanlife/findadoc-frontend#189]
It's a good practice to not use css classes in your html. Instead, you should use id's or classes that are name after what the element actually is. like `#cancellation-table-header` or `page-title`.

Bad Example
```
<div class="table table-header md-5 fluid">
...
<style scoped lang="scss">
  nothing....
</style>
```

Good Example
```
<div class="cancellations-table-header">
...
<style scoped lang="scss">
  .cancellations-table-header {
      @extend .table, .table-header, .md-5, .fluid;
  }
</style>
```

Going a step further, Ideally any reusable classes should be converted to placeholders as well. `%md-5, %fluid`.  This gives the ability to "tree-shake" unused shared classes. 

### Acceptance Criteria 

- [ ] Migrate each component to use sass styling
- [ ] Migrate css styling classes out of html (this may need to be split into multiple PR/issues)
- [ ] Going the extra mile: Replace shared css classes with placeholders.

**Comment by Al-un** [comment]
A sidenote: as styling is scoped, SCSS placeholders added value is pretty limited so using SCSS mixins actually has the same result

**Comment by Ermish** [comment]
> A sidenote: as styling is scoped, SCSS placeholders added value might be pretty limited so using SCSS mixins actually has the same result.
> 
> SCSS placeholders extracts common styling parts and group them to avoid styling redundancy. However, as styling is scoped by components, there is possibility of sharing styling so styling ends up being duplicated.

that's correct for css within the component. 
The key benefit for placeholders is for shared code outside the component level.  If it's not used, it's excluded from the build.

**Comment by claudiofreitas** [comment]
I made an attempt at this issue, but without success.
Went on `Alert.vue` then add to the bottom:
```
<style scoped lang="scss">
.something {
  @extend .grow;
}
</style>
```

Then the build process in the terminal shows:
```
 ERROR  in ./components/Alert.vue?vue&type=style&index=0&id=a3d5b1de&scoped=true&lang=scss&

Module build failed (from ./node_modules/extract-css-chunks-webpack-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: The target selector was not found.
Use "@extend .grow !optional" to avoid this error.
   ╷
30 │   @extend .grow;
   │   ^^^^^^^^^^^^^
   ╵
  components/Alert.vue 30:3  root stylesheet
```

I tried to investigate and found this thread:
https://github.com/vuetifyjs/vuetify/issues/12646

I could not solve yet.
I don't have too much knowledge on scss, I started using it recently,
so maybe someone more experienced can solve quicker.

Hope the info above helps somehow.

**Comment by Al-un** [comment]
@claudiofreitas ,
Without digging much into the problem, I agree with [this comment](https://github.com/vuetifyjs/vuetify/issues/12646#issuecomment-730241301). You might not want to extend such classes as it will create duplicate as long as `@import` syntax is used. Apparently, `@use` syntax is supposed to remove the problem but I am not familiar enough to tell.

As for why this will create some duplicates, it is down to how SCSS handles import. In the `Alert.vue`, the SCSS is standalone and the compiler tells "Hey, I don't know `.grow`". To do so, you need to "prepend" the SCSS file which contains the `.grow` definition. That's usually done by passing a parameter to the webpack sass-loader. In a plain vue.js application, updating `vue.config.js` is enough:

```js
// vue.config.js
module.exports = {
  // stuff

  // https://cli.vuejs.org/guide/css.html#passing-options-to-pre-processor-loaders
  css: {
    loaderOptions: {
      scss: {
        // sass-loader^9.0.0: additionalData
        // sass-loader^8.0.0: prependData
        // sass-loader before 8: data
        additionalData: `@import "<path to the Vuetify main SCSS file>";`,
      },
    },
  },

  // more stuff
}
```

But this means that the Vuetify main SCSS file will be defined multiple times: once in the `main.js` and once in each components. And that's where it ends up building duplicate. 

And...that's where my knowledge ends :p

**Comment by softwaredeveloptam** [comment]
Thanks for sharing your knowledge and progress @claudiofreitas and @Al-un
I'm not super knowledgeable about the subject but I'll try my best here as well 👍 

On exploration of the topics above, I believe we might be able to configure the `nuxt.config.js` under the `build` portion to minify the css so that if we decide to @import in individual vue components, we can minimize the impact to production of duplicate css classes.

Also, I'm assuming what @ermish is talking about is that we can extend vuetify's style options into our assets folder and make our own style guide.
https://github.com/vuetifyjs/vuetify/tree/master/packages/vuetify/src/styles/settings

Example:

**nuxt.config.js**
```js
// imports

export default {
  // other config
  css: [
    "@/assets/_variables.scss",
    // extending any other global scss files
  ]
  build: {
    extend(config, {isDev, isClient, isServer}) {
      // config webpack here with any specific rules we need under config
    }
  }
}
```

I'm thinking we can get all the css/frontend tooling out of the way if we can integrate a central styling system.
If we use storybook for contributing frontend devs or cypress for testing, we can maintain a consistent way for frontend developers to contribute code to the project as well.

Please correct me if I'm wrong anywhere 👍  🙏

**Comment by Al-un** [comment]
@softwaredeveloptam ,

You might want to check ([the nuxt setup](https://vuetifyjs.com/en/features/sass-variables/#nuxt-install))

If I may share additional links which could help you with styling fun:

- [CSS Specificity explained in MDN][3]
- [Vue.js scoped styling][4]
----

Sharing my opinion but obviously, leads' decision clearly takes precedence!

> TL;DR: I think it is fine to use Vuetify CSS classes but obviously we can extend the Vuetify CSS classes.

As found in a Vuetify example ([code here][1] and [rendering here][2]), it is perfectly fine to use Vuetify CSS classes so I think the current usage is OK. I understand the customization need . E.g, when overriding a `VCard`, I would then expect the component to look like:

```vue
<template>
  <div class="my-super-component">
    <v-card class="info-card fad-highlight elevated-4 mb-4 darken-1">
      ....
    </v-card>
  </div>
</template>

<script>
// <some folder>/MySuperComponent.vue
export default {
  name: 'MySuperComponent',
}
</script>

<style lang="scss" scoped>
// Personal convention: root element CSS class name matches component name
.my-super-component {
  // v-deep due to scoped styling
  ::v-deep .info-card {
    // your override here
  }
}
</style>
```

`.fad-highlight` (`fad` for "Find a Doc") being a common extension, it would be defined in some `styles/<...>/global.scss`:

```scss
// <...>/global.scss
.fad-highlight {
  // stuff
}
```

Otherwise, the other option is to have all styling done in vanilla SCSS files so the `<style>` section will not be required anymore.

[1]: https://github.com/vuetifyjs/vuetify/blob/master/packages/docs/src/examples/wireframes/inbox.vue#L17
[2]: https://vuetifyjs.com/en/examples/wireframes/inbox/
[3]: https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity
[4]: https://vue-loader.vuejs.org/guide/scoped-css.html#mixing-local-and-global-styles

**Comment by ermish** [comment]
Just a heads up, this ticket is focused around migrating our classes and css styling in html to sass. I believe @LaShawn (Project Lead) is already working on this one.

For actually upgrading to vuetify's sass mode, there's a separate ticket #137

**Comment by claudiofreitas** [comment]
Thanks everyone for the inputs!
My knowledge is low in this area, and since LaShawn is probably already working on this, I will search other tickets :D

**Comment by ermish** [comment]
After some coordinating, I'll try to take a look at this ticket and see if I can help ⚡

**Comment by ermish** [comment]
Closing this as we've decided to migrate to tailwind css.
